spring-cloud-dataflow-core/src/test/java/org/springframework/cloud/dataflow/core/dsl/StreamParserTests.java

```diff
@@ -339,73 +340,53 @@ public void expressions_gh1_2() {
+		checkForParseError(":boo. xx > file", DSLMessage.NO_WHITESPACE_IN_CHANNEL_DEFINITION, 6);
+		checkForParseError(":boo.xx. yy > file", DSLMessage.NO_WHITESPACE_IN_CHANNEL_DEFINITION, 9);
+		checkForParseError(":boo.xx .yy > file", DSLMessage.NO_WHITESPACE_IN_CHANNEL_DEFINITION, 8);
+		checkForParseError(":boo.xx . yy > file", DSLMessage.NO_WHITESPACE_IN_CHANNEL_DEFINITION, 8);
 
-		// 'queue' looks like a module all by itself so everything after is unexpected
-		checkForParseError("queue :bar > file", DSLMessage.NO_WHITESPACE_IN_CHANNEL_DEFINITION, 6);
-
-		checkForParseError("tap:queue: boo > file", DSLMessage.NO_WHITESPACE_IN_CHANNEL_DEFINITION, 11);
-		checkForParseError("tap:queue :boo > file", DSLMessage.NO_WHITESPACE_IN_CHANNEL_DEFINITION, 10);
-		checkForParseError("tap:queue : boo > file", DSLMessage.NO_WHITESPACE_IN_CHANNEL_DEFINITION, 10);
-
-		checkForParseError("tap:stream:boo .xx > file", DSLMessage.NO_WHITESPACE_IN_CHANNEL_DEFINITION, 15);
-		checkForParseError("tap:stream:boo . xx > file", DSLMessage.NO_WHITESPACE_IN_CHANNEL_DEFINITION, 15);
-		checkForParseError("tap:stream:boo. xx > file", DSLMessage.NO_WHITESPACE_IN_CHANNEL_DEFINITION, 16);
-		checkForParseError("tap:stream:boo.xx. yy > file", DSLMessage.NO_WHITESPACE_IN_CHANNEL_DEFINITION, 19);
-		checkForParseError("tap:stream:boo.xx .yy > file", DSLMessage.NO_WHITESPACE_IN_CHANNEL_DEFINITION, 18);
-		checkForParseError("tap:stream:boo.xx . yy > file", DSLMessage.NO_WHITESPACE_IN_CHANNEL_DEFINITION, 18);
-
-		checkForParseError("tap:queue:boo.xx.yy > file", DSLMessage.ONLY_A_TAP_ON_A_STREAM_OR_TASK_CAN_BE_INDEXED, 13);
-
-		sn = parse("wibble: http > queue:bar");
-		assertEquals("[((Label:wibble) ModuleNode:http)>(queue:bar)]", sn.stringify());
-	}
-
-	@Test
-	public void qualifiedSinkChannelError() {
-		// Only the source channel supports a dotted suffix
-		checkForParseError("http > queue:wibble.foo", DSLMessage.CHANNEL_INDEXING_NOT_ALLOWED, 19);
+		sn = parse("wibble: http > :bar");
+		assertEquals("[((Label:wibble) ModuleNode:http)>(bar)]", sn.stringify());
 	}
 
 	@Test
 	public void sourceChannel2() {
 		parse("foo = http | bar | file");
-		StreamNode ast = parse("tap:stream:foo.bar > file");
-		assertEquals("[(tap:stream:foo.bar:0>18)>(ModuleNode:file:21>25)]", ast.stringify(true));
-		assertEquals("tap:stream:foo.bar", ast.getSourceChannelNode().getChannelName());
+		StreamNode ast = parse(":foo.bar > file");
+		assertEquals("[(foo.bar:1>8)>(ModuleNode:file:11>15)]", ast.stringify(true));
+		assertEquals("foo.bar", ast.getSourceChannelNode().getChannelName());
 	}
 
 	@Test
 	public void sourceTapChannel() {
-		StreamNode ast = parse("tap:queue:xxy > file");
-		assertEquals("[(tap:queue:xxy:0>13)>(ModuleNode:file:16>20)]", ast.stringify(true));
+		StreamNode ast = parse(":xxy > file");
+		assertEquals("[(xxy:1>4)>(ModuleNode:file:7>11)]", ast.stringify(true));
 	}
 
 	@Test
 	public void sourceTapChannel2() {
```

Comment:
following from my comment above, I believe we can delete the existing `sourceTapChannel` and rename this one to `sourceTapChannel`.